ordered_set.py

```diff
@@ -1,45 +1,51 @@
 """
 An OrderedSet is a custom MutableSet that remembers its order, so that every
-entry has an index that can be looked up.
+entry has an index that can be looked up. It can also act like a Sequence.
 
 Based on a recipe originally posted to ActiveState Recipes by Raymond Hettiger,
 and released under the MIT license.
 """
 import itertools as it
-from collections import deque
-
-try:
-    # Python 3
-    from collections.abc import MutableSet, Sequence
-except ImportError:
-    # Python 2.7
-    from collections import MutableSet, Sequence
+from typing import (
+    Any,
+    Dict,
+    Iterable,
+    Iterator,
+    List,
+    MutableSet,
+    Sequence,
+    Set,
+    Type,
+    TypeVar,
+    Union,
+    overload,
+)
 
 SLICE_ALL = slice(None)
-__version__ = "3.1"
+__version__ = "4.0"
+
+
+T = TypeVar("T")
+SetLike = Union[Sequence[T], Set[T]]
 
 
-def is_iterable(obj):
+def is_atomic(obj: Any) -> bool:
```

Comment:
Was this a public API you expect users to use or only an internal helper? I'm not familiar with this function. 

If public, consider mentioning in the release notes that this was changed.

Reply:
If it's not public, then I would consider adding an underscore to state that.